plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayer.java

```diff
@@ -338,53 +351,56 @@ private void setLocationPoint(Point locationPoint) {
   // Styling
   //
 
+  private void styleForeground(@NonNull LocationLayerOptions options) {
+    boolean isGpsRenderMode = renderMode == RenderMode.GPS;
+    if (options.foregroundName() == null) {
```

Comment:
Can we have separate cases for `foregroundName` and `foregroundGpsName`? This would allow for providing an id for both normal foreground icon, as well as a GPS one.